src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -173,74 +172,62 @@ private bool CanUseVar(ITypeSymbol typeSymbol)
-                    var indexerDeclaration = ((IndexerDeclarationSyntax)newParentingNode);
-                    newParentingNode = indexerDeclaration
-                        .WithAccessorList(accessorList)
+                        .WithTriviaFrom(accessor);
+                case BaseMethodDeclarationSyntax baseMethod:
+                    return baseMethod
+                        .WithExpressionBody(null)
+                        .WithBody(body)
                         .WithSemicolonToken(SyntaxFactory.Token(SyntaxKind.None))
-                        .WithTrailingTrivia(indexerDeclaration.SemicolonToken.TrailingTrivia);
-                }
-            }
-            else if (oldParentingNode is BaseMethodDeclarationSyntax baseMethod)
-            {
-                newParentingNode = baseMethod.RemoveNode(oldBody, SyntaxRemoveOptions.KeepNoTrivia)
-                                             .WithBody(newBody);
-
-                if (newParentingNode.IsKind(SyntaxKind.MethodDeclaration))
-                {
-                    var methodDeclaration = ((MethodDeclarationSyntax)newParentingNode);
-                    newParentingNode = methodDeclaration
-                        .WithSemicolonToken(SyntaxFactory.Token(SyntaxKind.None))
-                        .WithTrailingTrivia(methodDeclaration.SemicolonToken.TrailingTrivia);
-                }
-                else if (newParentingNode.IsKind(SyntaxKind.OperatorDeclaration))
-                {
-                    var operatorDeclaration = ((OperatorDeclarationSyntax)newParentingNode);
-                    newParentingNode = operatorDeclaration
-                        .WithSemicolonToken(SyntaxFactory.Token(SyntaxKind.None))
-                        .WithTrailingTrivia(operatorDeclaration.SemicolonToken.TrailingTrivia);
-                }
-                else if (newParentingNode.IsKind(SyntaxKind.ConversionOperatorDeclaration))
-                {
-                    var conversionOperatorDeclaration = ((ConversionOperatorDeclarationSyntax)newParentingNode);
-                    newParentingNode = conversionOperatorDeclaration
+                        .WithTriviaFrom(baseMethod);
+                case LocalFunctionStatementSyntax localFunction:
+                    return localFunction
+                        .WithExpressionBody(null)
+                        .WithBody(body)
                         .WithSemicolonToken(SyntaxFactory.Token(SyntaxKind.None))
-                        .WithTrailingTrivia(conversionOperatorDeclaration.SemicolonToken.TrailingTrivia);
-                }
+                        .WithTriviaFrom(localFunction);
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(node);
             }
-
-            var newRoot = document.Root.ReplaceNode(oldParentingNode, newParentingNode);
-            return document.Document.WithSyntaxRoot(newRoot);
         }
 
         private async Task<Document> IntroduceLocalDeclarationIntoBlockAsync(
```

Comment:
from this point down these changes are also completely unrelated to the original issue